### PR TITLE
Add arm-hv feature in order to avoid page table permission problem when booting from el2

### PR DESCRIPTION
**Pull request by 915604903T**

### CODE REVIEW

File: page_table_entry/src/arch/aarch64.rs [hunk 1]
```diff
@@ -161,9 +161,12 @@ impl From<MappingFlags> for DescriptorAttr {
                 attr |= Self::UXN;
             }
         } else {
-            attr |= Self::UXN;
-            if !flags.contains(MappingFlags::EXECUTE) {
-                attr |= Self::PXN;
+            #[cfg(not(feature = "arm-hv"))]
+            {
+                attr |= Self::UXN;
```
Comment (hky1999) [inline page_table_entry/src/arch/aarch64.rs:166]: use XN bit for EL2.

See ARM® Architecture Reference Manual
-D4 The AArch64 Virtual Memory System Architecture
--D4.5 Access controls and memory region attributes
---Access permissions for instruction execution

File: page_table_entry/src/arch/aarch64.rs [hunk 2]
```diff
@@ -161,9 +161,12 @@ impl From<MappingFlags> for DescriptorAttr {
                 attr |= Self::UXN;
             }
         } else {
-            attr |= Self::UXN;
-            if !flags.contains(MappingFlags::EXECUTE) {
-                attr |= Self::PXN;
+            #[cfg(not(feature = "arm-hv"))]
```
Comment (hky1999) [inline page_table_entry/src/arch/aarch64.rs:164]: set feature name as "arm-el2"

File: page_table_entry/Cargo.toml [hunk 3]
```diff
@@ -11,6 +11,9 @@ repository.workspace = true
 keywords.workspace = true
 categories.workspace = true
 
+[features]
+arm-hv = []
```
Comment (hky1999) [inline page_table_entry/Cargo.toml:15]: set feature name as "arm-el2"

**Review by hky1999 — CHANGES_REQUESTED** [review]

**Review by equation314 — APPROVED** [review]
Please rebase commits into one.

File: page_table_entry/src/arch/aarch64.rs [hunk 4]
```diff
@@ -36,8 +36,11 @@ bitflags::bitflags! {
         const NG =          1 << 11;
         /// Indicates that 16 adjacent translation table entries point to contiguous memory regions.
         const CONTIGUOUS =  1 <<  52;
-        /// The Privileged execute-never field.
-        const PXN =         1 <<  53;
+        #[cfg(not(feature = "arm-el2"))]
```
Comment (equation314) [inline page_table_entry/src/arch/aarch64.rs:39]: Conditional compilation is not required here.

File: page_table_entry/src/arch/aarch64.rs [hunk 5]
```diff
@@ -36,8 +36,11 @@ bitflags::bitflags! {
...
+        {
+            /// The Privileged execute-never field.
+            const PXN =         1 <<  53;
+        }
         /// The Execute-never or Unprivileged execute-never field.
         const UXN =         1 <<  54;
```
Comment (equation314) [inline page_table_entry/src/arch/aarch64.rs:45]: Do we need a name that won't confuse? `UXN_OR_XN`? @hky1999

File: Cargo.toml [hunk 6]
```diff
@@ -7,7 +7,7 @@ members = [
 ]
 
 [workspace.package]
-version = "0.3.1"
+version = "0.3.2"
```
Comment (equation314) [inline Cargo.toml:10]: Do not include the commit "Fix rust docs, Bump to v0.3.2" here.

File: page_table_entry/src/arch/aarch64.rs (hunk 5 above)
Comment (hky1999) [reply page_table_entry/src/arch/aarch64.rs]: Will it be too complicated for mapping flags?

**Review by equation314 — APPROVED** [review]

**Comment by equation314** [comment]
Please fix CI

**Pull request merged**